lib/event_icon_builder.rb

```diff
@@ -0,0 +1,36 @@
+class EventIconBuilder
```

Comment:
But I definitely agree that this could be cleaned up structure wise to more closely resemble a json view folder structure. Will give this another pass.